### PR TITLE
test(userspace/libsinsp): add cases for formatters and minor improvements

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

/area tests

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

I realized that we don't have coverage in our output formatting system, which is also the construct used by Falco for formatting output strings and fields when security rules are triggered. This PR achieves decent coverage, and also adds minor improvements to the test suite for what concerns filters.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: userspace/libsinsp/test/sinsp_with_test_input.cpp [hunk 1]
```diff
@@ -255,10 +255,18 @@ sinsp_evt* sinsp_with_test_input::generate_proc_exit_event(int64_t tid_to_remove
 
 sinsp_evt* sinsp_with_test_input::generate_random_event(int64_t tid_caller)
 {
-	// Generate a random event on init to trigger the removal after proc exit
+	// Generate a random failed event. Useful when we want to trigger some actions on an event
+	// and we don't care about the event chosen.
```
Comment (LucaGuerra) [inline userspace/libsinsp/test/sinsp_with_test_input.cpp:259]: While you are at it could you change this function name, such as `generate_getcwd_failed_entry_event()`? Fortunately the event is not random

File: userspace/libsinsp/test/eventformatter.ut.cpp [hunk 2]
```diff
@@ -21,23 +21,240 @@ limitations under the License.
...
+		add_default_init_thread();
+		open_inspector();
+	}
+
+	void TearDown() override
+	{
+		m_last_res = false;
+		m_last_output.clear();
+		m_last_field_names.clear();
+		m_last_field_values.clear();
```
Comment (LucaGuerra) [inline userspace/libsinsp/test/eventformatter.ut.cpp:57]: Why is this necessary?

Comment (LucaGuerra) [reply userspace/libsinsp/test/eventformatter.ut.cpp:57]: Is the fixture maintained across all tests?

File: userspace/libsinsp/test/eventformatter.ut.cpp [hunk 3]
```diff
@@ -21,23 +21,240 @@ limitations under the License.
...
+	{
+		sinsp_evt_formatter f(&m_inspector, fmt, m_filter_list);
+		auto evt = generate_getcwd_exit_event();
+		f.get_field_names(m_last_field_names);
+		auto r1 = f.resolve_tokens(evt, m_last_field_values);
+		auto r2 = f.tostring_withformat(evt, m_last_output, of);
+		m_last_res = r1 && r2;
+	}
+
+	bool m_last_res;
```
Comment (LucaGuerra) [inline userspace/libsinsp/test/eventformatter.ut.cpp:71]: It's not really used uninitialized but let's initialize it anyways

**Comment by LucaGuerra** [comment]
Thank you! Just some minor comments

**Comment by jasondellaluce** [comment]
Thanks Luca! Accepted all your suggestions

**Review by Andreagit97 — APPROVED** [review]
/approve

**Comment by FedeDP** [comment]
@jasondellaluce can you rebase to fix ci?

**Review by LucaGuerra — APPROVED** [review]
